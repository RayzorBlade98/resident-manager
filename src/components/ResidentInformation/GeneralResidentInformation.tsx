import React from 'react';
import { MonthYearUtils } from '_/types/date';
import { Resident } from '_/types/resident';
import { convertCurrencyCentsToString } from '_/utils/currency';

interface GeneralResidentInformationProps {
  /**
   * Resident for which the information should be displayed
   */
  resident: Resident;
}

/**
 * Component that displays general information about a resident
 */
function GeneralResidentInformation(
  props: GeneralResidentInformationProps,
): JSX.Element {
  return (
    <>
      <p>{`${props.resident.firstName} ${props.resident.lastName}`}</p>
      <p>{`${convertCurrencyCentsToString(props.resident.rent[0].rent)}`}</p>
      <p>{`${MonthYearUtils.toString(props.resident.invoiceStart)}`}</p>
    </>
  );
}

export default GeneralResidentInformation;
