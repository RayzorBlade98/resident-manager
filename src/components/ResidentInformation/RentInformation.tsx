import { Icon } from '@mui/material';
import { MDBTable, MDBTableBody, MDBTableHead } from 'mdb-react-ui-kit';
import React from 'react';
import { v4 as uuid } from 'uuid';
import { Resident } from '_/types/resident';
import { convertCurrencyCentsToString } from '_/utils/currency';

interface RentInformationProps {
  /**
   * Resident for which the information should be displayed
   */
  resident: Resident;
}

/**
 * Component that displays rent information about a resident
 */
function RentInformation(props: RentInformationProps): JSX.Element {
  return (
    <MDBTable hover>
      <MDBTableHead>
        <tr>
          <th scope="col">Monat</th>
          <th scope="col">Miete</th>
          <th scope="col">Bezahlt</th>
          <th scope="col">Aktionen</th>
        </tr>
      </MDBTableHead>
      <MDBTableBody>
        {[...props.resident.rent].reverse().map((rent) => (
          <tr key={uuid()}>
            <td>{`${rent.dueDate.month} ${rent.dueDate.year}`}</td>
            <td>{convertCurrencyCentsToString(rent.rent)}</td>
            <td>
              {rent.isPaid ? (
                <Icon
                  baseClassName="fa-regular"
                  className="fa-circle-check"
                  color="success"
                />
              ) : (
                <Icon
                  baseClassName="fa-regular"
                  className="fa-circle-xmark"
                  color="error"
                />
              )}
            </td>
            <td />
          </tr>
        ))}
      </MDBTableBody>
    </MDBTable>
  );
}

export default RentInformation;
