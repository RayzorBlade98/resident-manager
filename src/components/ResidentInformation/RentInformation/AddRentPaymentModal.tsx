import {
  Button, Grid, InputAdornment, TextField,
} from '@mui/material';
import React, { useState } from 'react';
import GenericModal from '../../GenericComponents/GenericModal/GenericModal';
import { DateInputString, DateString, DateUtils } from '_/types/date';
import { CurrencyInCents, convertCurrencyEurosToCents } from '_/utils/currency';
import {
  ValidationError,
  ValidationErrorMessages,
  createValidationFunction,
} from '_/utils/validation';

export interface AddRentPaymentModalProps {
  show: boolean;

  /**
   * Callback when the modal is closed
   */
  onClose: () => void;

  /**
   * Callback when the payment input gets submitted
   * @param paymentAmount Amount that was paid
   * @param paymentDate Date on which the payment was received
   */
  onSave: (paymentAmount: CurrencyInCents, paymentDate: DateString) => void;
}

interface RentPaymentInput {
  paymentAmount: CurrencyInCents | null;
  paymentDate: DateString;
}

/**
 * Modal that provides functionality to insert payment information
 */
function AddRentPaymentModal(props: AddRentPaymentModalProps): JSX.Element {
  const [payment, setPayment] = useState<RentPaymentInput>({
    paymentAmount: null,
    paymentDate: DateUtils.getCurrentDate(),
  });
  const [errors, setErrors] = useState<
  ValidationErrorMessages<RentPaymentInput>
  >({});

  const validatePayment = createValidationFunction<RentPaymentInput>({
    paymentAmount: [ValidationError.Null, ValidationError.LessEqualZero],
  });

  function paymentUpdater(field: string) {
    function updatePayment(event: React.ChangeEvent<HTMLInputElement>): void {
      let value: number | string = event.target.value;
      if (field === 'paymentAmount') {
        value = Number(value);
        value = convertCurrencyEurosToCents(value);
      } else if (field === 'paymentDate') {
        value = DateUtils.convertDateInputString(value as DateInputString);
      }
      const newPayment = {
        ...payment,
        [field]: value,
      };
      setPayment(newPayment);
      setErrors({
        ...errors,
        [field]: validatePayment(
          newPayment,
          field as keyof ValidationErrorMessages<RentPaymentInput>,
        ),
      });
    }
    return updatePayment;
  }

  function onSave(): void {
    const newErrors = validatePayment(
      payment,
    ) as ValidationErrorMessages<RentPaymentInput>;
    const numberOfErrors = Object.keys(newErrors).filter(
      (k) => !!newErrors[k as keyof ValidationErrorMessages<RentPaymentInput>],
    ).length;
    if (numberOfErrors > 0) {
      setErrors(newErrors);
      return;
    }
    props.onSave(payment.paymentAmount as CurrencyInCents, payment.paymentDate);
  }

  return (
    <GenericModal
      title="Neue Zahlung"
      show={props.show}
      onClose={props.onClose}
      size="sm"
    >
      {/* Body */}
      <form>
        <Grid container columnSpacing={2} rowSpacing={2}>
          <Grid item xs={12}>
            <TextField
              id="paymentAmount"
              label="Zahlungssumme"
              type="number"
              required
              onChange={paymentUpdater('paymentAmount')}
              InputProps={{
                endAdornment: <InputAdornment position="end">€</InputAdornment>,
              }}
              error={!!errors.paymentAmount}
              helperText={errors.paymentAmount || ''}
            />
          </Grid>
          <Grid item xs={12}>
            <TextField
              id="paymentDate"
              label="Zahlungsempfang"
              type="date"
              required
              defaultValue={DateUtils.convertToDateInputString(
                payment.paymentDate,
              )}
              onChange={paymentUpdater('paymentDate')}
              InputLabelProps={{
                shrink: true,
              }}
              error={!!errors.paymentDate}
              helperText={errors.paymentDate || ''}
            />
          </Grid>
        </Grid>
      </form>
      {/* Footer */}
      <Button variant="contained" onClick={() => onSave()}>
        Hinzufügen
      </Button>
    </GenericModal>
  );
}

export default AddRentPaymentModal;
