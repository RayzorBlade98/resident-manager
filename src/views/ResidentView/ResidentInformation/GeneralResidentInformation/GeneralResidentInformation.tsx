import React from 'react';
import { useRecoilValue } from 'recoil';
import { convertCurrencyCentsToString } from '../../../../utils/currency/currency.utils';
import { convertNameToString } from '../../../../utils/name/name.utils';
import { residentViewSelectedResidentState } from '../../states/resident_view_state';
import { Resident } from '_/models/resident/resident';

/**
 * Component that displays general information about a resident
 */
function GeneralResidentInformation(): JSX.Element {
  const selectedResident = useRecoilValue(
    residentViewSelectedResidentState,
  ) as Resident;

  return (
    <>
      {selectedResident.contractResidents.map((contractResident) => (
        <p>{convertNameToString(contractResident.name, true)}</p>
      ))}
      <p>
        {convertCurrencyCentsToString(selectedResident.rentInformation[0].rent)}
      </p>
      <p>{selectedResident.contractStart.toString()}</p>
    </>
  );
}

export default GeneralResidentInformation;
