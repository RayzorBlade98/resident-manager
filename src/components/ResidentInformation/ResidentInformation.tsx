import { Resident } from '_/types/resident';
import { StyleAttribute } from 'glamor';
import {
  MDBTabs,
  MDBTabsContent,
  MDBTabsItem,
  MDBTabsLink,
  MDBTabsPane,
} from 'mdb-react-ui-kit';
import React, { useState } from 'react';
import GeneralResidentInformation from './GeneralResidentInformation';
import RentInformation from './RentInformation';

/**
 * Enum containing all tabs of this component
 */
enum ResidentTab {
  General = 'Information',
  Rent = 'Miete',
}

interface ResidentInformationProps {
  /**
   * Resident for which the information should be displayed
   */
  resident: Resident;

  /**
   * Style that should be applied to the container of the component. (Optional)
   */
  containerStyle: StyleAttribute;
}

/**
 * Component that displays information of the resident in different tabs.
 */
function ResidentInformation(props: ResidentInformationProps): JSX.Element {
  // Currently selected tab
  const [activeTab, setActiveTab] = useState<ResidentTab>(ResidentTab.General);

  /**
   * Handles the tab selection
   * @param tab selected tab
   */
  function onClickTab(tab: ResidentTab): void {
    if (tab === activeTab) return;
    setActiveTab(tab);
  }

  /**
   * Returns the content of the currently active tab
   */
  function getTabContent(): JSX.Element {
    switch (activeTab) {
      case ResidentTab.General:
        return <GeneralResidentInformation resident={props.resident} />;
      case ResidentTab.Rent:
        return <RentInformation resident={props.resident} />;
    }
  }

  return (
    <div {...props.containerStyle}>
      <MDBTabs fill className="mb-3">
        {Object.values(ResidentTab).map((tab: ResidentTab) => (
          <MDBTabsItem>
            <MDBTabsLink
              onClick={() => onClickTab(tab)}
              active={activeTab === tab}
            >
              {tab}
            </MDBTabsLink>
          </MDBTabsItem>
        ))}
      </MDBTabs>

      <MDBTabsContent>
        <MDBTabsPane show={true}>
          <div>{getTabContent()}</div>
        </MDBTabsPane>
      </MDBTabsContent>
    </div>
  );
}

export default ResidentInformation;
