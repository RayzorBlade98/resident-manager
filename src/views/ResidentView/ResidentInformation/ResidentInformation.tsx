import HomeIcon from '@mui/icons-material/Home';
import InfoIcon from '@mui/icons-material/Info';
import { Box, Tab, Tabs } from '@mui/material';
import React, { useState } from 'react';
import GeneralResidentInformation from './GeneralResidentInformation/GeneralResidentInformation';
import RentInformation from './RentInformation/RentInformation';

/**
 * Enum containing all tabs of this component
 */
enum ResidentTab {
  General = 0,
  Rent = 1,
}

/**
 * Component that displays information of the resident in different tabs.
 */
function ResidentInformation(): JSX.Element {
  const [activeTab, setActiveTab] = useState<ResidentTab>(ResidentTab.General);

  return (
    <>
      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
        <Tabs
          variant="fullWidth"
          value={activeTab}
          onChange={(_, tab: ResidentTab) => setActiveTab(tab)}
        >
          <Tab icon={<InfoIcon />} iconPosition="start" label="Informationen" />
          <Tab icon={<HomeIcon />} iconPosition="start" label="Miete" />
        </Tabs>
      </Box>
      <div hidden={activeTab !== ResidentTab.General}>
        <GeneralResidentInformation />
      </div>
      <div hidden={activeTab !== ResidentTab.Rent}>
        <RentInformation />
      </div>
    </>
  );
}

export default ResidentInformation;
