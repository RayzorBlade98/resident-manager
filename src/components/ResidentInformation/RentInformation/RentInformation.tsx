import CheckCircleOutlineIcon from '@mui/icons-material/CheckCircleOutline';
import HighlightOffIcon from '@mui/icons-material/HighlightOff';
import PaymentsOutlinedIcon from '@mui/icons-material/PaymentsOutlined';
import { Tooltip } from '@mui/material';
import { MDBTable, MDBTableBody, MDBTableHead } from 'mdb-react-ui-kit';
import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import AddRentPaymentModal from './AddRentPaymentModal';
import styles from './styles';
import { ResidentStateManager } from '_/states/saveStates/resident_state';
import { MonthYear, MonthYearUtils } from '_/types/date';
import { PaymentStatus, RentInformationUtils } from '_/types/rent';
import { Resident } from '_/types/resident';
import {
  CurrencyInCents,
  convertCurrencyCentsToString,
} from '_/utils/currency';
import { dateToString } from '_/utils/date';

interface RentInformationProps {
  /**
   * Resident for which the information should be displayed
   */
  resident: Resident;
}

/**
 * Component that displays rent information about a resident
 */
function RentInformation(props: RentInformationProps): JSX.Element {
  const [showModal, setShowModal] = useState<boolean>(false);
  const [modalDueDate, setModalDueDate] = useState<MonthYear>(
    MonthYearUtils.getCurrentMonthYear(),
  );

  return (
    <>
      {showModal && (
        <AddRentPaymentModal
          show={showModal}
          onClose={() => setShowModal(false)}
          onSave={(paymentAmount: CurrencyInCents, paymentDate: Date) => {
            // eslint-disable-next-line max-len
            ResidentStateManager.updateRentInformation(
              props.resident.id,
              modalDueDate,
              {
                paymentAmount,
                paymentDate,
              },
            );
            setShowModal(false);
          }}
        />
      )}
      <MDBTable hover>
        <MDBTableHead>
          <tr>
            <th scope="col">Monat</th>
            <th scope="col">Miete</th>
            <th scope="col">Nebenkosten</th>
            <th scope="col">Bezahlt</th>
            <th scope="col">Aktionen</th>
          </tr>
        </MDBTableHead>
        <MDBTableBody>
          {[...props.resident.rent].reverse().map((rent) => (
            <tr key={uuid()}>
              <td>{`${rent.dueDate.month} ${rent.dueDate.year}`}</td>
              <td>{convertCurrencyCentsToString(rent.rent)}</td>
              <td>{convertCurrencyCentsToString(rent.incidentals)}</td>
              <td>
                {RentInformationUtils.getPaymentStatus(rent)
                  === PaymentStatus.Paid && (
                  <Tooltip
                    // eslint-disable-next-line max-len
                    title={`Bezahlt am ${dateToString(rent.paymentDate as Date)}`}
                    arrow
                  >
                    <CheckCircleOutlineIcon color="success" />
                  </Tooltip>
                )}
                {RentInformationUtils.getPaymentStatus(rent)
                  === PaymentStatus.Unpaid && (
                  <Tooltip title="Unbezahlt" arrow>
                    <HighlightOffIcon color="error" />
                  </Tooltip>
                )}
                {RentInformationUtils.getPaymentStatus(rent)
                  === PaymentStatus.PaidPartially && (
                  <Tooltip
                    // eslint-disable-next-line max-len
                    title={`Teilweise bezahlt am ${dateToString(
                      rent.paymentDate as Date,
                    )} (${convertCurrencyCentsToString(
                      rent.paymentAmount as CurrencyInCents,
                    )} von ${convertCurrencyCentsToString(
                      RentInformationUtils.getAmountToPay(rent),
                    )})`}
                    arrow
                  >
                    <CheckCircleOutlineIcon color="warning" />
                  </Tooltip>
                )}
              </td>
              <td>
                {RentInformationUtils.getPaymentStatus(rent)
                  === PaymentStatus.Unpaid && (
                  <Tooltip title="Zahlung hinzufÃ¼gen" arrow>
                    <PaymentsOutlinedIcon
                      {...styles.actionIcon}
                      onClick={() => {
                        setModalDueDate(rent.dueDate);
                        setShowModal(true);
                      }}
                    />
                  </Tooltip>
                )}
              </td>
            </tr>
          ))}
        </MDBTableBody>
      </MDBTable>
    </>
  );
}

export default RentInformation;
