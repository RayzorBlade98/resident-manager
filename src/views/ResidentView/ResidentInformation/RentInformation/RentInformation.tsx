import CheckCircleOutlineIcon from '@mui/icons-material/CheckCircleOutline';
import HighlightOffIcon from '@mui/icons-material/HighlightOff';
import PaymentsOutlinedIcon from '@mui/icons-material/PaymentsOutlined';
import { Tooltip } from '@mui/material';
import { MDBTable, MDBTableBody, MDBTableHead } from 'mdb-react-ui-kit';
import React from 'react';
import { useRecoilValue, useSetRecoilState } from 'recoil';
import { v4 as uuid } from 'uuid';
// eslint-disable-next-line max-len
import { residentViewSelectedResidentState } from '../../states/resident_view_state';
import styles from '../../styles';
import AddRentPaymentModal from './AddRentPaymentModal/AddRentPaymentModal';
import addRentPaymentState from './states/add_rent_payment_state';
import { PaymentStatus, RentInformationUtils } from '_/types/rent';
import { Resident } from '_/types/resident';
import {
  CurrencyInCents,
  convertCurrencyCentsToString,
} from '_/utils/currency';
import { dateToString } from '_/utils/date';

/**
 * Component that displays rent information about a resident
 */
function RentInformation(): JSX.Element {
  const selectedResident = useRecoilValue(
    residentViewSelectedResidentState,
  ) as Resident;
  const setRentPaymentState = useSetRecoilState(addRentPaymentState);

  return (
    <>
      <AddRentPaymentModal />
      <MDBTable hover>
        <MDBTableHead>
          <tr>
            <th scope="col">Monat</th>
            <th scope="col">Miete</th>
            <th scope="col">Nebenkosten</th>
            <th scope="col">Bezahlt</th>
            <th scope="col">Aktionen</th>
          </tr>
        </MDBTableHead>
        <MDBTableBody>
          {[...selectedResident.rent].reverse().map((rent) => (
            <tr key={uuid()}>
              <td>{`${rent.dueDate.month} ${rent.dueDate.year}`}</td>
              <td>{convertCurrencyCentsToString(rent.rent)}</td>
              <td>{convertCurrencyCentsToString(rent.incidentals)}</td>
              <td>
                {RentInformationUtils.getPaymentStatus(rent)
                  === PaymentStatus.Paid && (
                  <Tooltip
                    // eslint-disable-next-line max-len
                    title={`Bezahlt am ${dateToString(
                      rent.paymentDate as Date,
                    )}`}
                    arrow
                  >
                    <CheckCircleOutlineIcon color="success" />
                  </Tooltip>
                )}
                {RentInformationUtils.getPaymentStatus(rent)
                  === PaymentStatus.Unpaid && (
                  <Tooltip title="Unbezahlt" arrow>
                    <HighlightOffIcon color="error" />
                  </Tooltip>
                )}
                {RentInformationUtils.getPaymentStatus(rent)
                  === PaymentStatus.PaidPartially && (
                  <Tooltip
                    // eslint-disable-next-line max-len
                    title={`Teilweise bezahlt am ${dateToString(
                      rent.paymentDate as Date,
                    )} (${convertCurrencyCentsToString(
                      rent.paymentAmount as CurrencyInCents,
                    )} von ${convertCurrencyCentsToString(
                      RentInformationUtils.getAmountToPay(rent),
                    )})`}
                    arrow
                  >
                    <CheckCircleOutlineIcon color="warning" />
                  </Tooltip>
                )}
              </td>
              <td>
                {RentInformationUtils.getPaymentStatus(rent)
                  === PaymentStatus.Unpaid && (
                  <Tooltip title="Zahlung hinzufÃ¼gen" arrow>
                    <PaymentsOutlinedIcon
                      {...styles.residentInformation.actionIcon}
                      onClick={() => {
                        setRentPaymentState((state) => ({
                          ...state,
                          selectedRentMonth: rent.dueDate,
                          showModal: true,
                        }));
                      }}
                    />
                  </Tooltip>
                )}
              </td>
            </tr>
          ))}
        </MDBTableBody>
      </MDBTable>
    </>
  );
}

export default RentInformation;
