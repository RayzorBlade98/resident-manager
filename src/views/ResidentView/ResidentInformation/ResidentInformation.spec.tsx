import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import * as GeneralResidentInformation from './GeneralResidentInformation/GeneralResidentInformation';
import * as RentInformation from './RentInformation/RentInformation';
import ResidentInformation from './ResidentInformation';

describe('ResidentInformation', () => {
  beforeAll(() => {
    jest
      .spyOn(GeneralResidentInformation, 'default')
      .mockReturnValue(<div>GeneralResidentInformation</div>);

    jest
      .spyOn(RentInformation, 'default')
      .mockReturnValue(<div>RentInformation</div>);
  });

  test.each([
    ['GeneralResidentInformation', 0],
    ['RentInformation', 1],
  ])(
    'should match snapshot (selected %s tab)',
    (tabName: string, tabId: number) => {
      // Arrange
      const renderResult = render(<ResidentInformation />);

      // Act
      const tab = renderResult.getAllByRole('tab').at(tabId)!;
      fireEvent.click(tab);

      // Assert
      expect(renderResult.container).toMatchSnapshot();
    },
  );
});
