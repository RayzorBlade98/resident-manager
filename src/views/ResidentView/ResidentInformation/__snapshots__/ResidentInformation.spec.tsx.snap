// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResidentInformation should match snapshot (selected GeneralResidentInformation tab) 1`] = `
<div>
  <div
    class="MuiBox-root css-1gsv261"
  >
    <div
      class="MuiTabs-root css-1ujnqem-MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-labelIcon MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-1nppm96-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTab-iconWrapper css-i4bv87-MuiSvgIcon-root"
              data-testid="InfoIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
            Informationen
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-labelIcon MuiTab-textColorPrimary MuiTab-fullWidth css-1nppm96-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTab-iconWrapper css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            Miete
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
  </div>
  <div>
    <div>
      GeneralResidentInformation
    </div>
  </div>
  <div
    hidden=""
  >
    <div>
      RentInformation
    </div>
  </div>
</div>
`;

exports[`ResidentInformation should match snapshot (selected RentInformation tab) 1`] = `
<div>
  <div
    class="MuiBox-root css-1gsv261"
  >
    <div
      class="MuiTabs-root css-1ujnqem-MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-labelIcon MuiTab-textColorPrimary MuiTab-fullWidth css-1nppm96-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTab-iconWrapper css-i4bv87-MuiSvgIcon-root"
              data-testid="InfoIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
            Informationen
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-labelIcon MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-1nppm96-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTab-iconWrapper css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            Miete
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
  </div>
  <div
    hidden=""
  >
    <div>
      GeneralResidentInformation
    </div>
  </div>
  <div>
    <div>
      RentInformation
    </div>
  </div>
</div>
`;
